### DISCUSSION TITLE
Problème - La transmission arrive plusieurs heures plus tard

### DISCUSSION THREAD

**Discussion by warmup79** [MTESSDev/FRW#288] [Annonces]
Bonjour,

Nous sommes à développer notre premier formulaire et nous éprouvons un problème avec la transmission. Lorsque nous utilisons l'outil de test de formulaire en ligne nous voyons que l'appel de notre service de transmission se fait immédiatement que nous utilisions le bouton "Tester transmission" ou  "Soumettre". 

Par contre lorsque nous créons le formulaire dans le contexte de notre application et qu'on appuie sur le bouton "Soumettre", ça prend plusieurs heures avant que notre service de transmission soit appelé. 

Ça ne semble pas être un problème de bastions car nous recevons des requêtes en provenance des 2 adresses IP possibles

Êtes-vous en mesure de voir pourquoi nous avons ce comportement.

Merci

**Comment by LarcM7** [comment]
Bonjour, 

Nous avons vu une erreur pour le formulaire Q05_Calendrier.

`{"No":"500","Message":"Invalid URI: The hostname could not be parsed.","MessageAffiche":null,"Cible":"/api/v1/Formulaires/OrchestrerProductionFormulaire/40949"};`

Nous voyons que votre URL contient une partie variable :

`url: https://{{{DonneesFormulaire.ConfigurationInterne.Domaine}}}/Applicatif/Ent/SQc.Ent.DossierAPI/api/v1/Questionnaires`

Le système n'arrive pas à résoudre l'URL.

Je vois `"configurationinterne":{"domaine":"w700u001app22.uf.sig-uf/C0"}` dans le contenu de votre formulaire.

Pouvez-vous valider si le domaine que vous préremplissez dans le formulaire est valide?

Par ailleurs à titre informatif nous avons ouvert une issue pour remonter l'erreur plus clairement dans le futur :  #289

**Comment by warmup79** [comment]
Bonjour, 

Oui, pour preuve, quand je passe par le testeur sans le mode simulation ça fonctionne sans problème :
![image](https://github.com/user-attachments/assets/31019dde-87a5-47fe-9ee9-b237981b4d84)

**Reply by warmup79** [reply]
A titre d'exemple, dans le contexte de notre application, nous avons créé et transmis un formulaire le 10 septembre, et la transmission a été reçu par notre api le 16 septembre. 

![image](https://github.com/user-attachments/assets/0f6d5982-8181-48c6-bb34-c7482807ae45)

**Reply by anisite** [reply]
Bonjour, c'est étrange que ça ait pris 6 jours "rentrer" c'est p-e vos cas de tests qui ont été réutilisés ?

une chose est certaine, le data "pré-rempli" semble changer de "case" (majuscue/minuscule) 

cela plante: 
````json
{"configurationinterne":{"domaine":"w700u001app22.uf.sig-uf/C0"}}
````

cela fonctionne: 
````json
{"ConfigurationInterne":{"Domaine":"w700u001app22.uf.sig-uf/C0"}}
````

Est-ce que le pré-remplissage diffère entre vos tests ? (exemple un fait manuellement et un fait via le votre traitement?). 
Quelle url vous avez dans votre barre adresse quand vous êtes dans le formulaire en erreur ?

Merci

**Reply by warmup79** [reply]
Bonjour, 

effectivement, il y avait une manipulation du contenu du message qui affectait la casse entre la journalisation et le préremplissage (FRW111) de notre côté. Les ajustements nécessaires ont été fait et maintenant la transmission fonctionne. 

Merci
